testing/unittests/jlibtests.cpp

```diff
@@ -3611,4 +3614,144 @@ CPPUNIT_TEST_SUITE_REGISTRATION( JLibUnicodeTest );
+        setSecretMount(secretRoot);
+        setSecretTimeout(100); // Set the timeout so we can check it is working.
+
+        secretRoot.append(PATHSEPCHAR).append("testing"); // catgegory
+        CPPUNIT_ASSERT(recursiveCreateDirectory(secretRoot.str()));
+    }
+
+    void testUpdate()
+    {
+        initPath(); // secretRoot needs to be called for each test
+
+        CPPUNIT_ASSERT(!hasSecret("secret1"));
+        writeTestingSecret("secret1", "value", "secret1Value");
+        //Secret should not appear yet - null should be cached.
+        CPPUNIT_ASSERT(!hasSecret("secret1"));
+
+        Owned<ISyncedPropertyTree> secret2 = resolveSecret("testing", "secret2", nullptr, nullptr);
+        CPPUNIT_ASSERT(!secret2->isValid());
+        CPPUNIT_ASSERT(!secret2->isStale());
+
+        MilliSleep(50);
+        //Secret should not appear yet - null should be cached.
+        CPPUNIT_ASSERT(!hasSecret("secret1"));
+        CPPUNIT_ASSERT(!secret2->isValid());
+        CPPUNIT_ASSERT(!secret2->isStale());
+
+        MilliSleep(100);
+        //Secret should not be updated - enough time has passed
+        checkSecret("secret1", "value", "secret1Value");
+        CPPUNIT_ASSERT(!secret2->isValid());
+        CPPUNIT_ASSERT(secret2->isStale());
+
+        Owned<ISyncedPropertyTree> secret3 = resolveSecret("testing", "secret3", nullptr, nullptr);
+        unsigned version = secret3->getVersion();
+        CPPUNIT_ASSERT(!secret3->isValid());
+        CPPUNIT_ASSERT(!secret3->isStale());
+        writeTestingSecret("secret3", "value", "secret3Value");
+        CPPUNIT_ASSERT(!secret3->isValid());
+        CPPUNIT_ASSERT_EQUAL(version, secret3->getVersion());
+
+        //After sleep new value should not have been picked up
+        MilliSleep(50);
+        CPPUNIT_ASSERT(!secret3->isValid());
+        CPPUNIT_ASSERT_EQUAL(version, secret3->getVersion());
+
+        //After sleep new value should now have been picked up
+        MilliSleep(100);
+        checkSecret("secret3", "value", "secret3Value");
+        unsigned version2 = secret3->getVersion();
+        CPPUNIT_ASSERT(!secret3->isStale());
+        CPPUNIT_ASSERT(secret3->isValid());
+        CPPUNIT_ASSERT(version != version2);
+
+        writeTestingSecret("secret3", "value", nullptr);
+        CPPUNIT_ASSERT(!secret3->isStale());
+        CPPUNIT_ASSERT(secret3->isValid());
+        CPPUNIT_ASSERT_EQUAL(version2, secret3->getVersion());
+
+        MilliSleep(50);
+        CPPUNIT_ASSERT(!secret3->isStale());
+        CPPUNIT_ASSERT(secret3->isValid());
+        CPPUNIT_ASSERT_EQUAL(version2, secret3->getVersion());
+        checkSecret("secret3", "value", "secret3Value");
+
+        MilliSleep(100);
+1        CPPUNIT_ASSERT(secret3->isStale()); // Value has gone
+        CPPUNIT_ASSERT(secret3->isValid());
+        CPPUNIT_ASSERT_EQUAL(version2, secret3->getVersion());
+        checkSecret("secret3", "value", "secret3Value");
```

Comment:
Great for resiliency.  I suppose to forcefully clear a secret (i.e. for emergencies) we assign an empty value.

Reply:
Yes, I think setting to a blank string should do that.  I'll add that as another test..